File: LenguajesProgramación/ejercicio3/java/EjemploExamen.java
class Examen{
    String nombre;
    private double test;
    private double problemas;
    double nota;

    public Examen(String nombre,double test,double problemas){
        this.nombre=nombre;
        this.test=test;
        this.problemas=problemas;
        this.nota=0.5*test+0.5*problemas;
    }

    public void nota(){
        System.out.println("Nombre: "+nombre);
        System.out.println("Nota: "+nota);
    }
}

class Examen_práctico extends Examen{
    double práctica;
    double nota_con_práctica;

    public Examen_práctico(String nombre, double test, double problemas, double práctica){
        super(nombre,test,problemas);
        this.práctica=práctica;
        this.nota_con_práctica=0.8*nota+0.2*práctica;
    }

    public void nota(){
        System.out.println("Nombre: "+nombre);
        System.out.println("Nota: "+nota_con_práctica);
    }
}
public class class_java {
    public static void main(String[] args){
        Examen examen_l=new Examen("Luiscarlos",9,8.5);
        Examen_práctico examen_s=new Examen_práctico("Sebastián",5,7,5.5);
        examen_l.nota();
        examen_s.nota();
        }
}

File: LenguajesProgramación/ejercicio3/python/EjemploExamen.py
class Examen:
    def __init__(self,nombre,test,problemas):
        self.nombre=nombre
        self.test=test
        self.problemas=problemas
        self.nota=1/2*test+1/2*problemas
    
    def mostrar_nota(self):
        print(f"Nombre:{self.nombre}" )
        print(f"Nota: {self.nota}")

class Examen_práctico(Examen):
    def __init__(self,nombre,test,problemas,práctica):
        super().__init__(nombre,test,problemas)
        self.práctica=práctica
        self.nota_con_práctica=4/5*self.nota+1/5*práctica

    def mostrar_nota(self):
        print(f"Nombre:{self.nombre}" )
        print(f"Nota: {self.nota_con_práctica}")

examen_l=Examen("Luiscarlos",9,8.5)
examen_s=Examen_práctico("Sebastián",5,7,5.5)
examen_l.mostrar_nota()
examen_s.mostrar_nota()